dmriprep/workflows/dwi/util.py

```diff
@@ -292,3 +292,181 @@ def _normalize(in_file, newmax=2000, perc=98.0):
+    )
+    vols = nib.load(in_file).shape[-1]
+    index_lines = np.ones((vols,))
+    index_lines_reshape = index_lines.reshape(1, index_lines.shape[0])
+    np.savetxt(out_file, index_lines_reshape, fmt="%i")
+    return out_file
+
+
+def gen_acqparams(in_file, total_readout_time=0.05):
+    # Generate the acqp file for eddy
+    import os
+    from ... import config
+    from nipype.utils.filemanip import fname_presuffix
+
+    # Get the metadata for the dwi
+    layout = config.execution.layout
+    metadata = str(layout.get_metadata(in_file))
+
+    # Generate output file name
+    out_file = fname_presuffix(
+        in_file,
+        suffix="_acqparams.txt",
+        newpath=os.path.abspath("."),
+        use_ext=False,
+    )
+
+    # Dictionary for converting directions to acqp lines
+    acq_param_dict = {
+        "j": "0 1 0 %.7f",
+        "j-": "0 -1 0 %.7f",
+        "i": "1 0 0 %.7f",
+        "i-": "-1 0 0 %.7f",
+        "k": "0 0 1 %.7f",
+        "k-": "0 0 -1 %.7f",
+    }
+
+    # Get encoding direction from json
+    if metadata.get("PhaseEncodingDirection"):
+        pe_dir = metadata.get("PhaseEncodingDirection")
+    else:
+        pe_dir = metadata.get("PhaseEncodingAxis")
+
+    # Get readout time from json, use default of 0.05 otherwise
+    if metadata.get("TotalReadoutTime"):
+        total_readout = metadata.get("TotalReadoutTime")
+    else:
+        total_readout = total_readout_time
+
+    # Construct the acqp file lines
+    acq_param_lines = acq_param_dict[pe_dir] % total_readout
+
+    # Write to the acqp file
+    with open(out_file, "w") as f:
+        f.write(acq_param_lines)
+
+    return out_file
+
+
+def init_eddy_wf(name="eddy_wf"):
+    """
+    Create an eddy workflow for head motion distortion correction on the dwi.
+
+    Parameters
+    ----------
+    name : str
+        Name of workflow (default: ``eddy_wf``)
+
+    Inputs
+    ----------
+    dwi_file
+        dwi NIfTI file
+    dwi_mask
+        Skull-stripping mask of reference image
+    in_bvec
+        File containing bvecs of dwi
+    in_bval
+        File containing bvals of dwi
+
+    Outputs
+    -------
+    out_eddy :
+        The eddy corrected diffusion image.
+    out_rotated_bvecs :
+        Rotated bvecs for each volume after eddy.
+    """
+    from nipype.interfaces.fsl import Eddy, EddyQuad
+
+    inputnode = pe.Node(
+        niu.IdentityInterface(
+            fields=[
+                "dwi_file",
+                "dwi_mask",
+                "in_bvec",
+                "in_bval",
+            ]
+        ),
+        name="inputnode",
+    )
+
+    outputnode = pe.Node(
+        niu.IdentityInterface(
+            fields=[
+                "out_eddy",
+                "out_rotated_bvecs"
+            ]
+        ),
+        name="outputnode",
+    )
+
+    workflow = Workflow(name=name)
+
+    eddy = pe.Node(
+        Eddy(repol=True, cnr_maps=True, residuals=True, method="jac"),
+        name="fsl_eddy",
```

Comment:
```suggestion
        name="eddy",
```
convention is using the same string for the variable name of the node and the node's name